uwblueprint/childrens-aid-society
# Sohail/intake review button routing

## Notion ticket link

[Implement Review Case button routing](https://www.notion.so/uwblueprintexecs/Implement-Review-Case-button-routing-35888bfc94a04064ac5b26510f949b31)



## Implementation description
* implement functionality for review button to lead to review case details step of intake page
* move declaration of "step" and "createStep" states from IntakePage.tsx into newly created IntakeValueContext.tsx
* wrap routes in App with IntakeContext
* reset "step" state back to step 0 using useEffect() whenever page location is changed, except when page is changed to intake after review button is clicked


## Steps to test
1.  Click on Test Status Modal > "Review" button - should be redirected to intake page, with review case details step open (no information will be present at the review step for this ticket)
![image](https://github.com/uwblueprint/childrens-aid-society/assets/44615627/6f55b73c-1267-4502-b75a-15b948609e47)
![image](https://github.com/uwblueprint/childrens-aid-society/assets/44615627/8439f6c5-fe73-4bec-8162-5d60da2accc1)
![image](https://github.com/uwblueprint/childrens-aid-society/assets/44615627/71d98ff5-96b6-408b-87e2-0aa67002ac64)

2. Complete intake steps and submit case, should go through as normal
3. If you return to dashboard and create a new case, should start at first step and proceed as expected


## What should reviewers focus on?
* To see if logic of intake steps breaks (if certain steps are skipped, or if it doesn't start at the first step when creating a new case or going straight to '/intake'), want to make sure the "step" state is being reset properly

## Checklist
